ui/packages/catalog-ui-search/src/main/webapp/component/metacard-overwrite/metacard-overwrite.view.js

```diff
@@ -13,116 +13,311 @@
  *
  **/
 /*global require*/
-var Marionette = require('marionette')
-var _ = require('underscore')
-var $ = require('jquery')
-var template = require('./metacard-overwrite.hbs')
-var CustomElements = require('../../js/CustomElements.js')
-var store = require('../../js/store.js')
-var ConfirmationView = require('../confirmation/confirmation.view.js')
-var Dropzone = require('dropzone')
-var Common = require('../../js/Common.js')
-var OverwritesInstance = require('../singletons/overwrites-instance.js')
-
-function getOverwriteModel(view) {
-  return OverwritesInstance.get(view.model.get('metacard').id)
+const store = require('../../js/store.js')
+const ConfirmationView = require('../confirmation/confirmation.view.js')
+const Dropzone = require('dropzone')
+const OverwritesInstance = require('../singletons/overwrites-instance.js')
+import React from 'react'
+import ReactDOM from 'react-dom'
+import styled from '../../react-component/styles/styled-components'
+import { readableColor } from 'polished'
+import {
+  Button,
+  buttonTypeEnum,
+} from '../../react-component/presentation/button'
+import withListenTo from '../../react-component/container/backbone-container'
```

Comment:
I could be wrong about this, but I think for this to work properly, you need to also import WithBackboneProps and pass them into your component. There's a simple example of this [here](https://github.com/codice/ddf/blob/15f20494516fa01613593c0b4055c99bcf020a8e/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/routes-container/routes-container.tsx)

Reply:
This is only required in typescript, so it isn't actually needed in this case.